kernel/queue.c

```diff
@@ -57,12 +57,14 @@ void k_queue_init(struct k_queue *queue)
 	SYS_TRACING_OBJ_INIT(k_queue, queue);
 }
 
+#if !defined(CONFIG_POLL)
```

Comment:
Shouldn't we make CONFIG_POLL mandatory now as if this is not set, then we would not be able to cancel the pending work, which would be very confusing for the end user as some config would allow cancelling always and some not.